### ISSUE TITLE
assist++ user in choosing newest background imagery.

### ISSUE THREAD

**Issue by mfrasca** [openstreetmap/iD#8633]
iD is the first OSM editor a beginner mapper will see, and apart from StreetComplete it is possibly the tool by which beginner mappers start contributing to OSM.  one task that mappers have to perform is choosing the "best" background imagery on which to base their contributions.  iD could play a more proactive role in this choice.

as of now, iD has one default imagery, and it's left to the user to decide whether they are happy with it.  a beginner mapper will probably not even be aware of the fact that a choice is possible.  the "assist++" term I'm using in the title, that would mean iD checking the timestamp of the available imageries, and proposing the newest one among the default choices.

the reason why I'm proposing this is not at all purely academic.  
* in 2021 in Panamá we have Bing stuck at 2016, and we keep getting external mappers contributing based on that, or even correcting the newer edits taking our work back to 5 years ago.
* on hotosm there's projects still lingering, which suggest usage of Bing. as long as their owners do not intervene, mappers keep using the old imagery.
 
a smarter iD would select the newest imagery, and display an alert message, when the user selects a different one, something like my favourite newspaper does when I insist in reading old news
 ![image](https://user-images.githubusercontent.com/122869/128776990-04dfae0d-5ac2-4a41-bd59-b8fa2574e668.png)

with this feature implemented, #8596 would not have taken place.

**Comment by boothym** [comment]
What happens when only one of the imagery sources has a timestamp available for iD to check? That is the case for the issue that you linked to. 

If iD recommended Esri for that area then you'd run into the other problem of users mapping without fixing the background imagery offset.

**Comment by mfrasca** [comment]
> What happens when only one of the imagery sources has a timestamp available for iD to check?

that's a good point which I had obviously missed.  I just chose a spot at random, tile/14/7796/4528, grabbed the headers using `curl` and I'm comparing Bing, Maxar, Mapbox and Esri.   

Bing lies: Last-Modified: Sat, 02 Jan 2021 13:23:28 GMT (seems that acquisition date should be 2016-03)
Mapbox says: Last-Modified: Thu, 10 Dec 2020 02:37:14 GMT
Maxar lies: last-modified: Thu, 01 Jan 1970 00:00:00 GMT
Esri says nothing.

I do not have an answer, only the perplexity about the issue.  could we hint at map providers that they give the timestamp of the tile with their answer?  after all, we are an important partner, aren't we?

please let's stay on the date problem.  alignment is a different issue, also serious, but perpendicular to the one in question.

**Comment by boothym** [comment]
> please let's stay on the date problem. alignment is a different issue, also serious, but perpendicular to the one in question.

Of course. However the reason I mentioned alignment is that simply choosing the newest imagery for the user doesn't mean you have solved the problem or given them the "best" imagery to use. :)

**Comment by andrewharvey** [comment]
> that's a good point which I had obviously missed. I just chose a spot at random, tile/14/7796/4528, grabbed the headers using curl and I'm comparing Bing, Maxar, Mapbox and Esri.

Last-Modified header is not the acquisition date of the imagery. It should reflect when the tile image file was last modified, which could be when it was last processed or created.

AFAIK there is not standard for acquisition date, some providers have a metadata API endpoint, others have dates baked into a transparent raster tile overlay, others don't provide this, but knowledgeable mappers can give a rough date just by looking at the imagery.

editor-layer-index has a best flag, but that doesn't help when Bing might be best for some areas of Bing, but not globally. We might need another index to let users contribute circa dates for imagery from the global basemaps in their area.

**Comment by matkoniecz** [comment]
Sadly, we have no access to info when imagery was taken so this functionality is impossible to provide

**Comment by andrewharvey** [comment]
> Sadly, we have no access to info when imagery was taken so this functionality is impossible to provide

That's not entirely true, many editor-layer-index sources include start/end date fields. But I'm not sure there's much value in doing anything in the UI without taking into account the global sources vintage, which we don't know about.

**Comment by matkoniecz** [comment]
lets rephrase it: in many cases we have no access to age of imagery, so overall making this comparison is not possible

**Comment by tyrasd** [comment]
Just for reference: In ELI we do manually mark some layers as `best` for the area they cover, which is very useful in case there are many different sources available e.g. from different years of aerial imagery.
